src/amount.cpp

```diff
@@ -15,7 +15,7 @@ CFeeRate::CFeeRate(const CAmount& nFeePaid, size_t nBytes_)
     int64_t nSize = int64_t(nBytes_);
 
     if (nSize > 0)
-        nSatoshisPerK = nFeePaid * 1000 / nSize;
+        nSatoshisPerK = nFeePaid * PRECISION_MULTIPLIER / nSize;
```

Comment:
Yes, it seems that was it, https://github.com/bitcoin/bitcoin/pull/7728/commits/4cb815a8f7c5422a6c5f4d020f027aefe7ffe172 seems to solve it (although it feels wrong for PRECISION_MULTIPLIER not to be size_t).

I will definitely remove it (that's what I should have done the first you complained, instead of moving it around to keep discussing about it), but since half of the PR discussion ended up being around that constant, I don't care about waiting a little bit longer to finish discussing it (in case is used in a future PR replacing  #7731 uses it).